cmake/modules/AddSwift.cmake

```diff
@@ -1261,6 +1261,10 @@ function(_add_swift_library_single target name)
       "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFTLIB_SINGLE_SUBDIR}"
       "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/../lib/swift/${SWIFT_SDK_${SWIFTLIB_SINGLE_SDK}_LIB_SUBDIR}")
 
+  if(DEFINED ENV{SDKROOT})
```

Comment:
Can you add a comment here, explaining why you are doing this? I would just say something like it is useful to be able to override this in certain cases or something like that.

Reply:
This is expanded as part of 4827ed7d85645f208f4dd3a31c7d08189f200bb7

Reply:
My suggestion:

"In certain cases when building, the environment variable SDKROOT is set to override where the sdk root is located in the system. If that environment variable has been set by the user, respect it and add the specified SDKROOT directory to the library_search_directories so we are able to link against those libraries".

A few questions:

1. Shouldn't you be turning off default search paths as well?
2. Can you add a precondition check that the specified path exists?

Reply:
Thanks for the suggestion!

My answers:
1. this would be a good idea -- however since I don't have firm grasp on where this code is employed, I preferred to append the new path to minimize the chance of disruption
2. will do, thanks for this tip as well!

Reply:
Incorporated new feedback in 88a046c959fbd777e972226d69af7d714b8f7967